### ISSUE TITLE
attr converter with default value - false error

### ISSUE THREAD

**Issue by Funth0mas** [python/mypy#6533]
When using `attr` attribute with both `converter` and `default`, wrong expectations is applied on `default` type with mypy == 0.670

The valid code
```python
import attr
import typing

def to_int(x: typing.Any) -> int:
    return int(x)

@attr.s
class C:
    a: int = attr.ib(converter=to_int, default="1")
```

results in:
```
main.py:9: error: Incompatible types in assignment (expression has type "str", variable has type "int")
main.py:9: error: Argument "converter" has incompatible type "Callable[[Any], int]"; expected "Optional[Callable[[Any], str]]"
```
is not right, because the `converter` is applied on `default` value as well.

Of course the hotfix is to convert the default value as well.

**Comment by ilevkivskyi** [comment]
@euresti Could you please confirm if this is correct?

**Comment by euresti** [comment]
Looks like `converters` do apply to `default` values.  This might be a little hard to fix just because right now we do all the type inference starting from the `default`.  

Though this is technically an issue with the stubs and not the plugin. Can one move this issue to https://github.com/python-attrs/attrs?

**Comment by ilevkivskyi** [comment]
I jus tried new "Transfer issue" GitHub feature, and it looks like it doesn't work. So we need to just close this here and open a new one at the attrs repo. @Funth0mas could you please open a new issue at there?
